runtime/parachains/src/assigner_on_demand/mod.rs

```diff
@@ -0,0 +1,534 @@
+				Err(SpotTrafficCalculationErr::QueueSizeLargerThanCapacity) => {
+					log::debug!(
+						target: LOG_TARGET,
+						"Error calculating spot traffic: The queue size is larger than the queue capacity."
+					);
+				},
+				Err(SpotTrafficCalculationErr::Division) => {
+					log::debug!(
+						target: LOG_TARGET,
+						"Error calculating spot traffic: Arithmetic error during division, either division by 0 or over/underflow."
+					);
+				},
+			};
+			T::DbWeight::get().reads_writes(2, 0)
+		}
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Create a single on demand core order.
+		/// Will use the spot price for the current block and can be set to reap the account if needed.
+		///
+		/// Parameters:
+		/// - `origin`: The sender of the call, funds will be withdrawn from this account.
+		/// - `max_amount`: The maximum balance to withdraw from the origin to place an order.
+		/// - `para_id`: A `ParaId` the origin wants to provide blockspace for.
+		/// - `keep_alive`: Should the transfer from the origin use the existential deposit keep alive checks.
+		///
+		/// Errors:
+		/// - `InsufficientFunds`
+		/// - `InvalidParaId`
+		/// - `QueueFull`
+		/// - `SpotPriceHigherThanMaxAmount`
+		///
+		/// Events:
+		/// - `SpotOrderPlaced`
+		#[pallet::call_index(0)]
+		#[pallet::weight(<T as Config>::WeightInfo::place_order())]
+		pub fn place_order(
+			origin: OriginFor<T>,
+			max_amount: BalanceOf<T>,
+			para_id: ParaId,
+			keep_alive: bool,
+		) -> DispatchResult {
+			// Is the tx signed
+			let sender = ensure_signed(origin)?;
+
+			// Traffic always falls back to 1.0
+			let traffic = SpotTraffic::<T>::get();
+
+			let config = <configuration::Pallet<T>>::config();
+
+			// Calculate spot price
+			let spot_price: BalanceOf<T> = traffic
+				.saturating_mul_int(config.on_demand_base_fee.saturated_into::<BalanceOf<T>>());
+
+			// Is the current price higher than `max_amount`
+			ensure!(spot_price.le(&max_amount), Error::<T>::SpotPriceHigherThanMaxAmount);
+
+			let existence_requirement = match keep_alive {
+				true => ExistenceRequirement::KeepAlive,
+				false => ExistenceRequirement::AllowDeath,
+			};
+
+			// Charge the sending account the spot price
+			T::Currency::withdraw(
+				&sender,
+				spot_price,
+				WithdrawReasons::FEE,
+				existence_requirement,
+			)?;
+
+			let assignment = Assignment::new(para_id);
+
+			let res = Pallet::<T>::add_on_demand_assignment(assignment, QueuePushDirection::Back);
+
+			match res {
+				Ok(_) => {
+					Pallet::<T>::deposit_event(Event::<T>::OnDemandOrderPlaced {
+						para_id,
+						spot_price,
+					});
+					return Ok(())
+				},
+				Err(err) => return Err(err),
+			}
+		}
+	}
+}
+
+impl<T: Config> Pallet<T>
+where
+	BalanceOf<T>: FixedPointOperand,
+{
+	/// The spot price multiplier. This is based on the transaction fee calculations defined in:
+	/// https://research.web3.foundation/en/latest/polkadot/overview/2-token-economics.html#setting-transaction-fees
```

Comment:
```suggestion
	/// https://research.web3.foundation/Polkadot/overview/token-economics#setting-transaction-fees
```

The link has changed for some reason